File: Bit Manipulation/insertion.cpp
#include <stdio.h>
#include <iostream>

using namespace std;

unsigned long insert(unsigned long a, unsigned long b, int i, int j){
  return (b<<i)|(((a>>j)<<j)|((a<<i)>>i));
}

int main(){
  unsigned long a=1024;
  unsigned long b=19;
  int i=2;
  int j=6;

  printf("%lu\n",insert(a,b,i,j));

}

File: Bit Manipulation/insertion.java
public class insertion{

  static int insert(int a, int b, int i, int j){
      return (b<<i)|(((a>>j)<<j)|((a<<i)>>i));
  }

  public static void main(String[] args){
    int a=1024;
    int b=19;
    int i=2;
    int j=6;

    System.out.println(insert(a,b,i,j));
  }

}
